cli/lib/snapshots.ts

```diff
@@ -20,6 +24,25 @@ import {
 } from 'cli/lib/appdata';
 import { LoggerError } from 'cli/logger';
 
+const UPDATE_SNAPSHOTS_LOCKFILE_PATH = path.join(
+	getAppdataDirectory(),
+	'studio-update-snapshots.lock'
+);
+
+function lock( path: string, options: lockfile.Options ) {
+	return new Promise< void >( ( resolve, reject ) => {
+		lockfile.lock( path, options, ( err ) => {
+			if ( err ) {
+				reject( err );
+			} else {
+				resolve();
+			}
+		} );
+	} );
+}
```

Comment:
The type definitions for `util.promisify` don't play nice with functions that accept more than one argument.